### DISCUSSION TITLE
data locations

### DISCUSSION THREAD

**Discussion by salsypoly** [zed-industries/zed#20123] [Feature Requests]
Hi i am a security member of a company whichi would like to use ZED, i would like to know where datas related to a user are saved physically?
regards

**Comment by bajrangCoder** [comment]
Most of the data (files) are stored in the following locations:

- **macOS**:
  ```
  ~/Library/Application Support/Zed/
  ```
  
- **Linux**:
  ```
  ~/.local/share/zed/
  ```

**Comment by Omar-Elrefaei** [comment]
If anyone is here later
For flatpack on Linux it is `~/.var/app/dev.zed.Zed/` which is standard location. 

But more importantly I found out that most of user data currently lives in sqlite database at `$DATA_DIR/data/zed/db/0-stable/db.sqlite` (including text in unsaved buffers). It was pretty cool actually exploring the db in a tool such as https://inloop.github.io/sqlite-viewer. 

I was poking around because I suspected that I lost some text in an unsaved buffer. 

Tip: [ripgrep-all](https://github.com/phiresky/ripgrep-all) can search sqlite dbs as plain text.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
